app.rb

```diff
@@ -82,10 +126,10 @@ def saml_settings
     else
       settings_file = 'config/saml_settings_demo.yml'
     end
-    settings = OneLogin::RubySaml::Settings.new(YAML.load_file settings_file)
-    settings.certificate = File.read('config/demo_sp.crt')
-    settings.private_key = File.read('config/demo_sp.key')
-    settings
+    base_config = Hashie::Mash.new(YAML.load_file(settings_file))
+    base_config.certificate = File.read('config/demo_sp.crt')
+    base_config.private_key = File.read('config/demo_sp.key')
+    OneLogin::RubySaml::Settings.new(base_config)
```

Comment:
It was broken previously. I believe SAML payloads were not signed previously, because the logic within RubySaml was checking for hash keys-as-symbols.